justfile

```diff
@@ -62,13 +62,22 @@ install-denv:
 configure *CONFIG:
     denv cmake -B build -S .  -DADDITIONAL_WARNINGS=ON  -DENABLE_CLANG_TIDY=ON  {{ CONFIG }}
 
+# configure minimal option for faster compilation
+configure-quick *CONFIG: 
+    denv cmake -B build -S . {{ CONFIG }}
+
+# Configure with Address Sanitizer (ASAN) and  UndefinedBehaviorSanitizer (UBSan)
+configure-asan-ubsan: (configure-quick "-DENABLE_SANITIZER_UNDEFINED_BEHAVIOR=ON -DENABLE_SANITIZER_ADDRESS=ON")
+
 # This is the same as just configure but reports all (non-3rd-party) warnings as errors
 configure-force-error *CONFIG: (configure "-DWARNINGS_AS_ERRORS=ON" CONFIG)
 
 # This is an alternative compiler (clang) together with enabling the LTO sanitizer
 # This is very helpful but for now it optimizes away our module linking, use for testing only
 configure-clang-lto: (configure "-DENABLE_LTO=ON -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=clang") 
 
+configure-gdb: (configure-quick "-DCMAKE_BUILD_TYPE=Debug")
+
```

Comment:
I'm kind of a DRY fiend and so I'm imagining a "parent" and private recipe that the "child" and public recipes use. This would mean we'd have one place that defines how `cmake` is called within the denv and the rest are just passing different configure options.

Not sure if this is worth it though since the code that is being repeated is pretty simple and short.

```suggestion
# base configure command defining how cmake is called, private so only experts call it
[private]
configure-base *CONFIG:
    denv cmake -B build -S . {{ CONFIG }}

# default configure of build when developing
configure *CONFIG: (configure-base "-DADDITIONAL_WARNINGS=ON -DENABLE_CLANG_TIDY=ON" CONFIG)

# configure minimal option for faster compilation
configure-quick: (configure-base)

# configure with Address Sanitizer (ASAN) and  UndefinedBehaviorSanitizer (UBSan)
configure-asan-ubsan: (configure-base "-DENABLE_SANITIZER_UNDEFINED_BEHAVIOR=ON -DENABLE_SANITIZER_ADDRESS=ON")

# This is the same as just configure but reports all (non-3rd-party) warnings as errors
configure-force-error: (configure "-DWARNINGS_AS_ERRORS=ON")

# Use alternative compiler and enable LTO (test compiling only, won't run properly)
configure-clang-lto: (configure "-DENABLE_LTO=ON -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=clang") 

# Keep debug symbols so running with gdb provides more helpful detail
configure-gdb: (configure-base "-DCMAKE_BUILD_TYPE=Debug")
```